contrib/gitian-builder/gitian-build.sh

```diff
@@ -0,0 +1,428 @@
+  if [[ $result == "" ]]; then
+    echo "No GPG keys available..."
+    echo "Please follow this documentation: https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/generating-a-new-gpg-key"
+    exit 1
+  fi
+
+  # Check that a signer is specified
+  if [[ $SIGNER == "" ]]; then
+    echo "$scriptName: Missing signer."
+    echo "Try $scriptName --help for more information"
+    exit 1
+  fi
+fi
+
+if [[ $push == false && $setup == false ]]; then
+  # Check that a version is specified
+  if [[ $VERSION == "" ]]; then
+    echo "$scriptName: Missing version."
+    echo "Try $scriptName --help for more information"
+    exit 1
+  fi
+fi
+
+#######################
+######## SETUP ########
+#######################
+if [[ $setup == true ]]; then
+  git clone https://github.com/dogecoin/gitian.sigs.git
+  git clone https://github.com/dogecoin/dogecoin-detached-sigs.git
+  git clone https://github.com/devrandom/gitian-builder
+
+  pushd gitian-builder || exit 1
+
+  git fetch origin 9e97a4d5038cd61215f5243a37c06fa1734a276e # LAST VERSION TESTED
+  git reset --hard FETCH_HEAD
+  git am < "$dirName"/patches/0001-Docker-apt-cacher.patch
+
+   ./bin/make-base-vm --docker --arch amd64 --suite trusty || exit 1
+  "$dirName"/setup/dependencies.sh || exit 1
+
+  popd  || exit 1
+  exit
+
+fi
+
+function move_build_files() {
+  find build/out -type f -exec mv '{}' ../dogecoin-binaries/${VERSION}/ \;
+}
+
+function download_descriptor() {
+  if [[ ! $1 == 'test' ]]; then
+    uri="${url/github.com/raw.githubusercontent.com}"/"$2"/contrib/gitian-descriptors/gitian-"$1".yml
+    echo "Downloading descriptor ${1} ${uri}"
+    wget $uri -O gitian-"$1".yml || exit 1
+  else
+    # CI tests
+    cp ../ci/descriptor/"$1".yml gitian-"$1".yml || exit 1
+  fi
+}
+
+#############################
+######## DESCRIPTORS ########
+#############################
+# Make descriptors folder
+mkdir -p ./gitian-descriptors/
+
+pushd gitian-descriptors || exit 1
+
+if [[ $build == true || $verify == true ]]; then
+  for descriptor in "${DESCRIPTORS[@]}"; do
+    download_descriptor "$descriptor" "$COMMIT"
+  done
+fi
+
+if [[ $sign == true ]]; then
+  for sign_descriptor in "${SIGN_DESCRIPTORS[@]}"; do
+    download_descriptor "$sign_descriptor" "$COMMIT"
+  done
+fi
+
+popd
+
+#######################
+######## BUILD ########
+#######################
+if [[ $build == true ]]; then
+
+  # Make output folder
+  mkdir -p ./dogecoin-binaries/"$VERSION"
+
+  pushd ./gitian-builder || exit 1
+
+  # CLEAN dogecoin git directory because of old caching
+  if [[ -d inputs/dogecoin/ ]]; then
+    echo "Cleaning Dogecoin directory..."
+    rm -rf inputs/dogecoin/
+  fi
+
+  for descriptor in "${DESCRIPTORS[@]}"; do
+    echo ""
+    echo "Compiling ${VERSION} ${descriptor}"
+    echo ""
+    ./bin/gbuild -j "$proc" -m "$mem" --commit dogecoin="$COMMIT" --url dogecoin="$url" ../gitian-descriptors/gitian-"$descriptor".yml  || exit 1
```

Comment:
Don't know what did you do with all of your dependencies scripts, info were probably there, but looks like some are missing now.

Need `ruby` to run `gbuild`.